sources/packages/web/src/views/student/StudentRequestChange.vue

```diff
@@ -0,0 +1,166 @@
+          >
+        </div>
+      </full-page-container>
+    </div>
+    <!-- Appeal view-->
+    <div v-else>
+      <full-page-container>
+        <body-header
+          title="Fill in the field(s) below"
+          subTitle="StudentAid BC will review your application change after you submit the fields below."
+        ></body-header>
+        <student-appeal-form
+          v-for="formName in appealFormNames"
+          :key="formName"
+          :formName="formName"
+          @appealFormLoaded="appealFormLoaded"
+        ></student-appeal-form>
+        <!-- action area -->
+        <div class="mt-4">
+          <v-btn color="primary" outlined @click="backToRequest">Back</v-btn>
+          <v-btn
+            @click="submitAppeal"
+            class="primary-btn-background float-right"
+            >Submit</v-btn
+          >
+        </div>
+      </full-page-container>
+    </div>
+  </v-container>
+</template>
+<script lang="ts">
+import { computed, ref } from "vue";
+import {
+  StudentRequest,
+  StudentAppealRequestDTO,
+  StudentAppealDTO,
+} from "@/types";
+import { ApplicationService } from "@/services/ApplicationService";
+import { StudentAppealService } from "@/services/StudentAppealService";
+import formio from "@/components/generic/formio.vue";
+import FullPageContainer from "@/components/layouts/FullPageContainer.vue";
+import HeaderNavigator from "@/components/generic/HeaderNavigator.vue";
+import BodyHeader from "@/components/generic/BodyHeader.vue";
+import StudentAppealForm from "@/components/common/StudentAppealForm.vue";
+import { useToastMessage } from "@/composables";
+const INVALID_APPLICATION_NUMBER = "INVALID_APPLICATION_NUMBER";
+
+export default {
+  components: {
+    HeaderNavigator,
+    FullPageContainer,
+    formio,
+    BodyHeader,
+    StudentAppealForm,
+  },
+  setup() {
+    const toast = useToastMessage();
+    let requestFormData: any = undefined;
+    let applicationId: number;
+    const appealFormNames = ref([] as string[]);
+    let appealForms: any = [];
+    const showRequestForAppeal = computed(
+      () => appealFormNames.value.length === 0,
+    );
+
+    const formLoaded = (form: any) => {
+      requestFormData = form;
+    };
+
+    const submitRequest = async (data: StudentRequest) => {
+      try {
+        const application = await ApplicationService.shared.getApplicationForRequestChange(
+          data.applicationNumber,
+        );
+        applicationId = application.id;
+        appealFormNames.value = data.formNames;
+      } catch (error) {
+        let errorMessage = "An error happened while requesting a change.";
+        if (error.response.data?.errorType === INVALID_APPLICATION_NUMBER) {
+          errorMessage = error.response.data.message;
+        }
+        toast.error("Unexpected error", errorMessage);
+      }
+    };
+
+    const submitStudentRequest = async () => {
+      return requestFormData.submit();
+    };
+
+    const backToRequest = () => {
```

Comment:
the name and logic inside does make sense to me

Reply:
Doesn't make sense? yes, I get your feeling. This is a 2 stage request cycle where you enter the application number and select appeals in stage 1 and then move to stage 2 where you enter appeal specific details. Without v-stepper we are showing this with hide and view. So basically going back to previous view needs to re-initialize the values.